react/AvailabilitySubscriber.tsx

```diff
@@ -0,0 +1,209 @@
+import React, { useState } from 'react'
+import type { FormEvent, ChangeEvent } from 'react'
+import { useIntl } from 'react-intl'
+import { useMutation } from 'react-apollo'
+import { Button, Input } from 'vtex.styleguide'
+import useProduct from 'vtex.product-context/useProduct'
+
+import ADD_TO_AVAILABILITY_SUBSCRIBER_MUTATION from './graphql/addToAvailabilitySubscriberMutation.gql'
+import styles from './styles/availability-subscriber.css'
+
+interface Props {
+  /* Availability of the product */
+  available?: boolean
+  /* SKU id to subscribe to */
+  skuId?: boolean
+}
+
+interface CommertialOffer {
+  AvailableQuantity: number
+}
+
+const isAvailable = (commertialOffer?: CommertialOffer) => {
+  return (
+    commertialOffer &&
+    (Number.isNaN(+commertialOffer.AvailableQuantity) ||
+      commertialOffer.AvailableQuantity > 0)
+  )
+}
+
+/**
+ * Availability Subscriber Component.
+ * A form where users can sign in to be alerted
+ * when a product becomes available again
+ */
+function AvailabilitySubscriber(props: Props) {
+  const productContext = useProduct()
+  const [name, setName] = useState('')
+  const [email, setEmail] = useState('')
+  const [emailError, setEmailError] = useState(false)
+  const [didBlurEmail, setDidBlurEmail] = useState(false)
+
+  const [addToAvailabilitySubscriber, { loading, error, data }] = useMutation(
```

Comment:
How about renaming the mutation function to `signUp`? I don't think this name is very descriptive of what is actually done by the mutation